crates/rpc/src/v04/method/estimate_message_fee.rs

```diff
@@ -0,0 +1,42 @@
+use pathfinder_common::{BlockId, CallParam, ContractAddress, EntryPoint, EthereumAddress};
+use stark_hash::Felt;
+
+use crate::context::RpcContext;
+use crate::v02::method::call::FunctionCall;
+use crate::v02::types::reply::FeeEstimate;
+use crate::v03::method::estimate_message_fee::EstimateMessageFeeError;
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct EstimateMessageFeeInput {
+    message: MsgFromL1,
+    sender_address: EthereumAddress,
+    block_id: BlockId,
+}
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct MsgFromL1 {
+    pub from_address: EthereumAddress,
+    pub to_address: ContractAddress,
+    pub payload: Vec<CallParam>,
+}
+
+pub async fn estimate_message_fee(
+    context: RpcContext,
+    input: EstimateMessageFeeInput,
+) -> Result<FeeEstimate, EstimateMessageFeeError> {
+    let entry_point_selector = EntryPoint(
+        Felt::from_be_slice(input.message.from_address.0.as_bytes())
+            .expect("H160 bytes will not overflow felt"),
+    );
+    let input = crate::v03::method::estimate_message_fee::EstimateMessageFeeInput {
+        message: FunctionCall {
+            contract_address: input.message.to_address,
+            entry_point_selector,
+            calldata: input.message.payload,
+        },
```

Comment:
I'm not convinced this is correct?

Reply:
It looks OK on the first blink. Would be great to add a test though that makes sure we got this mapping right.

Reply:
`entry_point_selector` has been added, so maybe it's time to re-visit this PR.

Reply:
Fixed